packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -517,31 +525,29 @@ class _ScaffoldLayout extends MultiChildLayoutDelegate {
     }
 
     Rect floatingActionButtonRect;
-    if (hasChild(_ScaffoldSlot.floatingActionButton)) {
-      final Size fabSize = layoutChild(_ScaffoldSlot.floatingActionButton, looseConstraints);
-
-      // To account for the FAB position being changed, we'll animate between
-      // the old and new positions.
-      final ScaffoldPrelayoutGeometry currentGeometry = ScaffoldPrelayoutGeometry(
-        bottomSheetSize: bottomSheetSize,
-        contentBottom: contentBottom,
-        contentTop: contentTop,
-        floatingActionButtonSize: fabSize,
-        minInsets: minInsets,
-        scaffoldSize: size,
-        snackBarSize: snackBarSize,
-        textDirection: textDirection,
-      );
-      final Offset currentFabOffset = currentFloatingActionButtonLocation.getOffset(currentGeometry);
-      final Offset previousFabOffset = previousFloatingActionButtonLocation.getOffset(currentGeometry);
-      final Offset fabOffset = floatingActionButtonMotionAnimator.getOffset(
-        begin: previousFabOffset,
-        end: currentFabOffset,
-        progress: floatingActionButtonMoveAnimationProgress,
-      );
-      positionChild(_ScaffoldSlot.floatingActionButton, fabOffset);
-      floatingActionButtonRect = fabOffset & fabSize;
-    }
+    fabStatuses.forEach(
```

Comment:
Using a for loop here is recommended over `Iterable.forEach` since you're passing in a function literal.

https://dart.dev/guides/language/effective-dart/usage#avoid-using-iterableforeach-with-a-function-literal

Reply:
`fabStatuses` is a `Map<Key, _FabStatus>`; I don't know if `for` works in this case.

"Also note that it’s always OK to use Map.forEach(). Maps aren’t iterable, so this guideline doesn’t apply." https://dart.dev/guides/language/effective-dart/usage#avoid-using-iterableforeach-with-a-function-literal